threeML/plugins/OGIP/eventlist.py

```diff
@@ -1108,6 +1005,370 @@ def worker(start_index):
+
+        # Dead time correction
+
+        exposure = 0.
+        for tmin, tmax in zip(tmin_list, tmax_list):
+            exposure += tmax - tmin
+
+        if self._dead_time is not None:
+
+            total_dead_time = self._dead_time[time_mask].sum()
+        else:
+
+            total_dead_time = 0.
+
+        self._exposure = exposure - total_dead_time
+        # self._total_dead_time = total_dead_time
+
+        self._tmin_list = tmin_list
+        self._tmax_list = tmax_list
+
+        self._active_dead_time = total_dead_time
+
+
+
+class EventListWithLiveTime(EventList):
+    def __init__(self, arrival_times, energies, n_channels, live_time, live_time_starts, live_time_stops,
+                 start_time=None, stop_time=None,
+                 first_channel=0, rsp_file=None, ra=None, dec=None, mission=None, instrument=None):
+        """
+        Container for event style data which are tagged with time and energy/PHA.
+
+
+
+        :param  arrival_times: list of event arrival times
+        :param  energies: list of event energies or pha channels
+        :param live_time: array of livetime fractions
+        :param live_time_starts: start of livetime fraction bins
+        :param live_time_stops:  stop of livetime fraction bins
+        :param mission:
+        :param instrument:
+        :param  n_channels: Number of detector channels
+        :param  start_time: start time of the event list
+        :param  stop_time: stop time of the event list
+        :param  first_channel: where detchans begin indexing
+        :param  rsp_file: the response file corresponding to these events
+
+
+        :param  ra:
+        :param  dec:
+        """
+
+        EventList.__init__(self, arrival_times, energies, n_channels, start_time, stop_time, first_channel, rsp_file,
+                           ra, dec,
+                           mission, instrument)
+
+        assert len(live_time) == len(
+                live_time_starts), "Live time fraction (%d) and live time start (%d) have different shapes" % (
+            len(live_time), len(live_time_starts))
+
+        assert len(live_time) == len(
+                live_time_stops), "Live time fraction (%d) and live time stop (%d) have different shapes" % (
+            len(live_time), len(live_time_stops))
+
+        self._live_time = np.asarray(live_time)
+        self._live_time_starts = np.asarray(live_time_starts)
+        self._live_time_stops = np.asarray(live_time_stops)
+
+    def _exposure_over_interval(self, tmin, tmax):
+        """
+
+        :param tmin: start time of interval
+        :param tmax: stop time of interval
+        :return: exposure
+        """
+
+        # First see if the interval is completely contained inside a
+        # livetime interval. In this case we only compute that.
+
+
+
+
+        inside_idx = np.logical_and(self._live_time_starts < tmin, tmax < self._live_time_stops)
+
+        # see if it contains elements
+
+        if self._live_time[inside_idx]:
```

Comment:
Aha, I'm not doing searches, instead I'm building index filters... but if you want you have a better way let me know.